r/tools/test-nixlibs.R

```diff
@@ -157,6 +157,10 @@ test_that("check_allowlist", {
 })
 
 test_that("find_latest_nightly()", {
+  skip_if(
+    getRversion() > "4.4.0",
+    "long last version components (>8) fail to max on r-devel"
+  )
```

Comment:
I will send this to r-devel, but looks like running on _actual_ r-devel these checks have been failing. When the last component is over 8 digits, max/min etc. simply return the first element 🙃 

We also could consider taking one digit off of our padding, which would make this work. I tried looking for where we add those numbers in and got to https://github.com/apache/arrow/blob/acf7bdc4588fb41ba9a5f81e91056795850c9a8f/dev/archery/archery/crossbow/core.py#L856-L857 but not sure if just changing that to have one less `0` is the only thing that would need to be changed. 

In R-devel:
```
> max(package_version(c("1.0.1.100000000", "1.0.3.100000000", "1.0.2.100000000")))
[1] ‘1.0.1.100000000’
> max(package_version(c("1.0.1.10000000", "1.0.3.10000000", "1.0.2.10000000")))
[1] ‘1.0.3.10000000’
```

In 4.4.0:
```
> max(package_version(c("1.0.1.100000000", "1.0.3.100000000", "1.0.2.100000000")))
[1] ‘1.0.3.100000000’
> 
> max(package_version(c("1.0.1.10000000", "1.0.3.10000000", "1.0.2.10000000")))
[1] ‘1.0.3.10000000’
```

Reply:
We can make those numbers smaller now. We aren't doing calendar versioning anymore so we don't need to make sure these are bigger than those.

Reply:
I'll make an issue for doing that and do it separately from here. Would also be good to hear from @assignUser if going slightly smaller is ok at this point now?

Reply:
https://github.com/apache/arrow/issues/41415